upgrade/upgrade_common_services.sh

```diff
@@ -185,6 +187,46 @@ EOF
     fi
 }
 
+function compare_channel() {
```

Comment:
Please add a comment, sentence or 2 about the intent of this function. I can see its comparing channels, but why? Why is this function needed now when we didn't need it before 3.5?